### ISSUE TITLE
Drawer icon and drawer screen overlay not accessible for talkback

### ISSUE THREAD

**Issue by tiamed** [react-navigation/react-navigation#10621]
### Current behavior

Talkback can only recognize the drawer icon but not narrating it's label.
Talkback cannot find the overlay to go back when entered drawer content screen.

### Expected behavior

Accessibility label provided for drawer icon and drawer content screen overlay

### Reproduction

 https://snack.expo.io/?platform=android&name=Drawer%20Navigator%20%7C%20React%20Navigation&dependencies=%40expo%2Fvector-icons%40*%2C%40react-native-community%2Fmasked-view%40*%2Creact-native-gesture-handler%40*%2Creact-native-pager-view%40*%2Creact-native-paper%40%5E4.7.2%2Creact-native-reanimated%40*%2Creact-native-safe-area-context%40*%2Creact-native-screens%40*%2Creact-native-tab-view%40%5E3.0.0%2C%40react-navigation%2Fbottom-tabs%406.3.1%2C%40react-navigation%2Fdrawer%406.4.1%2C%40react-navigation%2Felements%401.3.3%2C%40react-navigation%2Fmaterial-bottom-tabs%406.2.1%2C%40react-navigation%2Fmaterial-top-tabs%406.2.1%2C%40react-navigation%2Fnative-stack%406.6.1%2C%40react-navigation%2Fnative%406.0.10%2C%40react-navigation%2Fstack%406.2.1&hideQueryParams=true&sourceUrl=https%3A%2F%2Freactnavigation.org%2Fexamples%2F6.x%2Fsimple-drawer.js

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [X] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [ ] @react-navigation/​material-top-tabs
- [ ] @react-navigation/​stack
- [ ] @react-navigation/​native-stack

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/drawer               | 6.4.1
| react-native                           |0.68.2
| expo                                   |45.0.0
| node                                   |v12.22.10
| npm or yarn                            |6.14.16
